lib/ast.ml

```diff
@@ -28,8 +28,10 @@ type stmt =
   | Block of stmt list
   | If of expr * stmt * stmt
   | While of expr * stmt
+  | For of expr * expr * expr * stmt
   | Return of expr
-
+  | No_op
```

Comment:
`No_op` as a statement does not make much sense to me. I will prefer to avoid those ambiguous definitions.

Reply:
thats not ambiguous, in fact CompactiC implements that as noexpr.

it is a placeholder, for something that doesnt exist. I have added an if statement in the semantic code that handles this. In fact exactly how microC (Stephen Edward's) handles it in the CFG, but I do it at the semantics time.

It will not in fact cause codegen issues, because I add in an empty SBlock, and codegen uses the sast.

I know this is for machine code, but this reference will explain how its just a place holder: 
https://www.noop.com.au/noop-meaning/

There are no ambiguities in the grammar, or else there would be shift/reduce or reduce/reduce conflicts. All of that is being handled

Reply:
`noexpr` is indeed needed to implement microc, but I believe they don't have `nostmt`, and it sounds not right to me. 

 I also don't think we need `noexpr` though. it's there in microc only because they support c's legacy grammar, i.e.
`for (;;) {}` and `return;`. 

But we don't have this kind of grammar, in viz, we are using `infinite_loop(){}` and `return none;`. no? 😄

I believe we can make it work in codegen, just seems redundant to me. Maybe we could ask others' opinions.